src/middlewares/authentication.ts

```diff
@@ -0,0 +1,74 @@
+import { Request, Response, NextFunction } from 'express';
+import jsonwebtoken from 'jsonwebtoken';
+import * as dotenv from 'dotenv';
+dotenv.config();
+
+import { User } from '../models/user';
+import { UserRoleEnum } from '../utils/common/enum';
+import CustomError from '../errors/custom-errors';
+interface IJWTPayload {
+  _id: string;
+}
+
+class HandleAuthentication {
+  public async userAuthentication(
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<Response | void> {
+    const authToken = req.header('Authorization')?.replace('Bearer ', '');
+    if (!authToken) {
+      next();
+      return;
+    }
+    const decoded = jsonwebtoken.verify(
+      authToken as string,
+      process.env.JWT_SECRET_KEY as string
+    );
+    const userId = decoded as IJWTPayload;
+
+    const user = await User.findOne({
+      _id: userId._id
+    });
+
+    if (!user) {
+      throw new CustomError('No User Found', 404);
+    }
+
+    if (user.enabled === false) {
+      throw new CustomError('User Account Disabled', 401);
+    }
+
+    req.authToken = authToken;
+    req.user = user;
```

Comment:
Don't send any errors here.

```suggestion
    if (user && user.enabled === true) {
        req.authToken = authToken;
        req.user = user;
    }
```